rewriter/tests/sighandler/main.c

```diff
@@ -47,19 +40,25 @@ void install_sighandler(struct handler *h) {
     sigaction(SIGTRAP, &sa, NULL);
 }
 
-void test_handler(void) {
+Test(sighandler, main_rewriter) {
+    install_sighandler(NULL);
     raise(SIGTRAP);
-    test_handler_from_lib();
 }
 
-int main() {
-    setbuf(stdout, NULL);
-    install_sighandler(NULL);
-    test_handler();
-
+Test(sighandler, main_ignore) {
     static struct handler h = {
         .handler = IA2_SIGHANDLER(trap_handler),
     };
     install_sighandler(&h);
```

Comment:
How is this any different from the NULL case? It seems to install the same signal handler? Maybe this used to behave differently, but it seems the same to me now...

Reply:
Maybe I misunderstood, but I thought the idea was to assemble the struct manually to make sure that the rewriter/sighandler macro was setting it up correctly. @ayrtonm thoughts?

Reply:
yeah @sim-immunant is right. The behavior at runtime is the same, but the idea was to make sure that `sa.sa_handler = $FN_PTR_EXPR` and `sa.sa_handler = $FN_IDENTIFIER` both work (in terms of both it builds and it runs). This seemed important since this field is one of those function pointer types that we don't rewrite (because the type is defined in a header outside the root directory), so I wanted to make sure the rewriter wasn't doing something wrong like rewriting the function pointer expressions or inserting call gates that shouldn't be there.